cmd/run/flags.go

```diff
@@ -7,191 +7,145 @@ import (
 
 // bindRunFlags binds the cobra cmd flags to the equivalent config value being managed
 // by viper. This bridges the config between cobra flags and viper flags.
-func bindRunFlags(command *cobra.Command) {
-	defaultConfig := DefaultConfig()
+func bindRunFlags(command *cobra.Command, _ []string) {
 	flags := command.Flags()
```

Comment:
I'm wondering why we duplicate the flags construction and why the flags don't live alongside the bindings here? Can't we pass in the flags at the caller layer and then return the PreRun func? Or is there an issue with that?

```go
func bindRunFlagFunc(flags *pflag.FlagSet) func(*cobra.Command, []string) {
    return func(cmd *cobra.Command, args []string {
        util.MustBindPFlag(..., flags.Lookup(...))
    }
}
```
and then it's usage 
```go
// cmd/run/run.go
flags := command.Flags()

cmd := &cobra.Command{
    ...
    PreRun: bindRunFlagsFunc(flags)
}
```

Reply:
`flags := command.Flags()` is referencing the created `cobra.Command`, i need to build that object first

Reply:
So then
```go
cmd := &cobra.Command{
    ...
}

flags := cmd.Flags()

cmd.PreRun = bindRunFlagsFunc(flags)
```